components/chef-ui-library/src/molecules/chef-modal/chef-modal.tsx

```diff
@@ -168,6 +177,26 @@ export class ChefModal {
     return '';
   }
 
+  // when Angular detects the modal is open
+  // it sets the focus by default on the close button for unlocked modals,
+  // or the modal div for locked modals
+  // Developer can specify element to focus first using firstFocus attribute
+  private getFocusElement(lockStatus: boolean): HTMLElement {
+    const modal = this.el.getElementsByClassName('modal').item(0) as HTMLElement;
+    const closeFocus = this.el.getElementsByClassName('close').item(0).firstElementChild as HTMLElement;
+    let firstFocus = this.el.querySelector('[firstFocus]') as HTMLElement;
+
+    if (lockStatus) {
+      return modal
+    } else if (firstFocus) {
+      return firstFocus.tagName === 'CHEF-BUTTON'
+        ? firstFocus.firstElementChild as HTMLElement
+        : firstFocus
+    }
+
+    return closeFocus;
+  }
```

Comment:
This function has been updated so that when unlocked, it's searching for the `firstFocus` attribute in the modal.  If it finds that attribute, it will also check if it is a `chef-button` and then apply focus to the correct element.

I needed to add an exception for `chef-button` because the actual button element is nested below the `chef-button` tag.

`firstFocus` will work with any of the native input elements but may need similar alteration for other `chef-`elements.  I don't believe we have a need for many though.  Often times we are looking simply for a text input or a 'cancel' button.

Reply:
It sounds like we could genericize this (without having to check for `chef-button` specifically) -- I am thinking that a native HTML input element would not have children (or only text node children), so that is how you could distinguish.

Reply:
Agreed - I would like to _not_ need to adjust in the future.  Checking for children is a good idea - i'll toy with that.

Reply:
I played with this but came to discover that other possible elements that might want to be focused first, like a `chef-select` will have child nodes, but should't focus the child.  

So as it sits, I think this is the best way to do this for our needs for the current expected implementation.